pkg/agent/baker.go

```diff
@@ -946,6 +954,129 @@ func isMariner(osSku string) bool {
 	return osSku == datamodel.OSSKUCBLMariner || osSku == datamodel.OSSKUMariner
 }
 
+const sysctlTemplateString = `# This is a partial workaround to this upstream Kubernetes issue:
+# https://github.com/kubernetes/kubernetes/issues/41916#issuecomment-312428731
+net.ipv4.tcp_retries2=8
+net.core.message_burst=80
+net.core.message_cost=40
+{{- if .CustomLinuxOSConfig}}
+  {{ if .CustomLinuxOSConfig.Sysctls}}
+    {{ if .CustomLinuxOSConfig.Sysctls.NetCoreSomaxconn -}}
+      net.core.somaxconn={{.CustomLinuxOSConfig.Sysctls.NetCoreSomaxconn}}
```

Comment:
I doubt this indentation works and in checking the decoded customdata from UT I think it's messed up

this needs to be de-dented all the way, or else you need to use hyphens `{{- foo -}}` to trim whitespace

Reply:
dedented it.